scylla.go

```diff
@@ -452,19 +450,19 @@ func (p *scyllaConnPicker) shardOf(token int64Token) int {
 	return int(sum >> 32)
 }
 
-func (p *scyllaConnPicker) Put(conn *Conn) {
+func (p *scyllaConnPicker) Put(conn *Conn) error {
 	var (
 		nrShards = conn.scyllaSupported.nrShards
 		shard    = conn.scyllaSupported.shard
 	)
 
 	if nrShards == 0 {
-		panic(fmt.Sprintf("scylla: %s not a sharded connection", p.address))
+		return fmt.Errorf("scylla: %s not a sharded connection", p.address)
```

Comment:
Let's change a message here:

```suggestion
		return errors.New("server reported that it has no shards", p.address)
```